### PR TITLE
12.0 ADD module mrp_request_bom_structure

### PR DESCRIPTION
**Pull request by bealdav**
Selection_123

cc @florian-dacosta @hparfr @LoisRForgeFlow

### CODE REVIEW

File: mrp_request_bom_structure/models/mrp_request.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    _inherit = "mrp.production.request"
+
+    def button_open_structure_report(self):
+        """ - store quantity in parameter
+            - return action client
+        """
+        self.ensure_one()
+        IConf = self.env["ir.config_parameter"]
+        param = IConf.get_param("request_bom_structure")
+        if not param or (param[0] != "{" and param[:-1] != "}"):
```
Comment (hparfr) [inline mrp_request_bom_structure/models/mrp_request.py]: try except https://docs.python.org/fr/3/library/json.html#json.JSONDecodeError ?

**Review by hparfr — CHANGES_REQUESTED** [review]

**Comment by LoisRForgeFlow** [comment]
wouldn't it be more "odooish" to place it under action dropdown?

**Comment by bealdav** [comment]
It could be an option, but my client paid me this dev to avoid clicks. Dropdown make one more click.
If all users of this module want to go towards dropdown button, I have to override the view to add the button to directly click on it.

**Comment by LoisRForgeFlow** [comment]
Got it, you are the only user for now so I guess you have the power :)

**Comment by bealdav** [comment]
@LoisRForgeFlow please could make a final review, thanks.

I fixed code and have a better test coverage now

**Review by dora-jurcevic — APPROVED** [review]
Feature tested and working nicely

**Review by thibaultrey — APPROVED** [review]
Functionally reviewed

**Review by marcelsavegnago — APPROVED** [review]

**Comment by bealdav** [comment]
Thanks a lot @dreispt

**Comment by bealdav** [comment]
Hi @dreispt could you try again ? Thanks

**Comment by bealdav** [comment]
please @anajuaristi could you merge ?

**Comment by bealdav** [comment]
please @dreispt could you try again ?

**Comment by bealdav** [comment]
@LoisRForgeFlow FYI, I observe a random issue on v12 branch with mrp_subcontracting

Selection_139

Other topic, I fixed travils file on #579

**Comment by bealdav** [comment]
Please @LoisRForgeFlow , now branch is completely fixed, could you try again ? ;-)
Thanks

**Comment by pedrobaeza** [comment]
Merging blindly as requested by @bealdav 

/ocabot merge nobump

**Pull request merged**